src/decoder/lattice-faster-online-decoder.h

```diff
@@ -337,12 +338,23 @@ class LatticeFasterOnlineDecoder {
 
   /// Processes emitting arcs for one frame.  Propagates from prev_toks_ to cur_toks_.
   /// Returns the cost cutoff for subsequent ProcessNonemitting() to use.
-  BaseFloat ProcessEmitting(DecodableInterface *decodable);
+  /// To reduce vtable lookup cost in openfst based graph search, 
+  //make ProcessEmitting() and ProcessNonemitting() be templated on the FstType,
```

Comment:
should be space between // and comment.

Reply:
... also the comment is too long.  And keep the triple ///, it relates to Doxygen.
 I'd just add a single line:
/// Templated on Fst type for speed; called via ProcessEmittingWrapper().